ATDevOutreach/DevFest-Abuja-Code-Challenge-2018
# Pusher: A service that allows users send emails by sending a sms to a AT's shortcode OR Dailing a USSD (Useful when there's no internet)

# What it does
This is an sms to email & ussd to email service, that allows users send mails in case they don't have internet connection. Useful for field workers in villages and areas where internet is bad.

e..g. User simply sends `Can you cancel all my meetings? receiver@gmail.com` to `28701` and this instantly sends a mail to *receiver@gmail.com* with the sms contents.

OR Dials a USSD Code such as `*384*4110#` and enters the text (containing the email ) such as:
`Can you cancel all my meetings? receiver@gmail.com` and this sends a mail to *receiver@gmail.com* with the USSD text contents.


## To Test Service:
- Create a Shortcode OR USSD Number from your AT's account
- For SMS: Add this URL as your Callback URL: `https://puzher.herokuapp.com/listener/`
- For USSD: Add this URL as your Callback URL: `https://puzher.herokuapp.com/ussd-listener/`
- Use the [sandbox](https://simulator.africastalking.com) to send SMS to your shortcode.
- SMS/USSD text should be inform of: `Can you cancel all my meetings? receiver@gmail.com` , the content of the mail will be sent to the mail included in your text.

## Screenshots:

![atsimulator](https://user-images.githubusercontent.com/11256481/48865279-6f880b00-edcf-11e8-98f7-53490b165ce4.PNG)

![email](https://user-images.githubusercontent.com/11256481/48865270-6860fd00-edcf-11e8-81cc-3135080ab687.PNG)

![email](https://user-images.githubusercontent.com/11256481/48853487-e6ada700-edaf-11e8-9ce6-1a3128733f24.PNG)


## Deployed [here](http://puzher.herokuapp.com/) | Local Setup Instructions: [here](https://github.com/Timtech4u/DevFest-Abuja-Code-Challenge-2018/blob/2348136457976/SETUP.md)
- Sign up / login with Phone Number and Password
- Retrieve the shortcode OR USSD OR steps to setup either on Africa Talking Dashboard
- Enjoy our *sms2mail* *ussd2mail* services 😀
